### ISSUE TITLE
Quarkus 3 : In native mode, Kafka streams extension doesn't support Snappy compression

### ISSUE THREAD

**Issue by Foricher** [quarkusio/quarkus#32962]
### Describe the bug

After upgrading from Quarkus 2.16.1.Final to Quarkus 3.0.1.Final in native mode,
an exception  java.lang.UnsatisfiedLinkError is thrown when kafka streams uses Snappy compression.
In JVM mode, there is no issue.

[quarkus3-kafka-snappy.log](https://github.com/quarkusio/quarkus/files/11344449/quarkus3-kafka-snappy.log)

2023-04-27 12:43:00,081 INFO  [com.ale.ovn.per.ser.DataPersistencyService] (main) The application is starting...
2023-04-27 12:43:00,081 INFO  [com.ale.ovn.per.ser.DataPersistencyService] (main) The application is starting done
2023-04-27 12:43:00,084 INFO  [io.quarkus] (main) ovng-datapersistency 1.0.0-SNAPSHOT native (powered by Quarkus 3.0.1.Final) started in 0.179s. Listening on: http://0.0.0.0:8080
2023-04-27 12:43:00,084 INFO  [io.quarkus] (main) Profile prod activated.
2023-04-27 12:43:00,084 INFO  [io.quarkus] (main) Installed features: [cdi, confluent-registry-avro, hibernate-validator, kafka-client, kafka-streams, logging-manager, micrometer, rest-client-reactive, rest-client-reactive-jackson, resteasy-reactive, resteasy-reactive-jsonb, scheduler, smallrye-context-propagation, smallrye-health, smallrye-openapi, smallrye-reactive-messaging, smallrye-reactive-messaging-kafka, swagger-ui, vertx]
2023-04-27 12:43:01,087 INFO  [org.apa.kaf.str.KafkaStreams] (pool-7-thread-1) stream-client [ovng_datapersistency-229cf6d1-46b7-4f7f-9404-5e5b5ee72578] State transition from CREATED to REBALANCING
2023-04-27 12:43:01,102 INFO  [org.apa.kaf.str.pro.int.GlobalStateManagerImpl] (ovng_datapersistency-229cf6d1-46b7-4f7f-9404-5e5b5ee72578-GlobalStreamThread) global-stream-thread [ovng_datapersistency-229cf6d1-46b7-4f7f-9404-5e5b5ee72578-GlobalStreamThread] Restoring state for global store DATA_PERSISTENCY_STATE_STORE
Exception in thread "ovng_datapersistency-229cf6d1-46b7-4f7f-9404-5e5b5ee72578-GlobalStreamThread" java.lang.UnsatisfiedLinkError: org.xerial.snappy.SnappyNative.uncompressedLength(Ljava/lang/Object;II)I [symbol: Java_org_xerial_snappy_SnappyNative_uncompressedLength or Java_org_xerial_snappy_SnappyNative_uncompressedLength__Ljava_lang_Object_2II]
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:53)
        at org.xerial.snappy.SnappyNative.uncompressedLength(SnappyNative.java)
        at org.xerial.snappy.Snappy.uncompressedLength(Snappy.java:638)
        at org.xerial.snappy.SnappyInputStream.hasNextChunk(SnappyInputStream.java:435)
        at org.xerial.snappy.SnappyInputStream.read(SnappyInputStream.java:466)
        at java.base@17.0.5/java.io.DataInputStream.readByte(DataInputStream.java:271)
        at org.apache.kafka.common.utils.ByteUtils.readUnsignedVarint(ByteUtils.java:170)
        at org.apache.kafka.common.utils.ByteUtils.readVarint(ByteUtils.java:205)

### Expected behavior

Kafka streams with Snappy compression works in native mode.

### Actual behavior

 An exception is thrown with kafka streams
 java.lang.UnsatisfiedLinkError: org.xerial.snappy.SnappyNative.uncompressedLength(Ljava/lang/Object;II)I

### Output of `uname -a` or `ver`

Linux ovng-data-persistency-94d879475-cvhvt 5.15.77-flatcar #1 SMP Tue Dec 6 22:49:50 -00 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08)

### GraalVM version (if different from Java)

quay.io/quarkus/ubi-quarkus-native-image:22.3-java17

### Quarkus version or git rev

3.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

**Comment by MarcAntoine-Beauvais** [comment]
I am having same issue with Snappy but for usage of parquet files.

But looking at the documentation, Snappy is not supported in native (for Kafka at least):
https://quarkus.io/guides/kafka#using-snappy-for-message-compression

Looking for a way to use Snappy in native...

**Comment by cescoffier** [comment]
I don't think snappy compression was supported in native in 2.x. 

Snappy requires a native library which can be a bit cumbersome to include in a native executable (need the right architecture, the right declaration, unpacking on app startup (which will increase application startup).

**Comment by cescoffier** [comment]
Let me rephrase. It works with the reactive messaging kafka connector. For kafka streams, I don't know the code path and how snappy gets accessed.

**Comment by sarathpetec** [comment]
> Sorry, i forgot to set quarkus.kafka.snappy.enabled=true when building in native mode. By adding this option and kafka-streams.producer.compression.type=snappy for kafka streams it works fine. So i can delete/close this issue.

This is just awesome
